assets/elm/Main.elm

```diff
@@ -153,14 +244,59 @@ update msg model =
         GetCollection ->
             model ! [ getCollection ]
 
+        GetNotes ->
+            model ! [ getNotes model ]
+
+        NewNotes (Ok notes) ->
+            let
+                _ =
+                    Debug.log "hi" (toString notes)
+            in
+                { model | notes = notes } ! []
+
+        NewNotes (Err e) ->
+            let
+                _ =
+                    Debug.log "NewNotes Err" (toString e)
+            in
+                model ! []
+
+        SearchInput search ->
+            let
+                newModel =
+                    { model | search = search }
+            in
+                newModel ! [ getNotes newModel ]
+
+        PageChange page ->
+            { model | page = page } ! []
+
+        PageChangeToSearch ->
+            let
+                newModel =
+                    { model | page = Search, search = "" }
+            in
+                newModel ! [ getNotes newModel ]
```

Comment:
I think this `newModel` pattern should be avoided and just write the new model twice